src/Hit/Issue.hs

```diff
@@ -164,3 +162,24 @@ parseOwnerRepo url =
 
     stripGitSuffix :: Text -> Maybe Text
     stripGitSuffix x = whenNothing (T.stripSuffix ".git" x) (Just x)
+
+data AssignSuccess = AlreadyAssigned | NewlyAssigned
+
+assignToIssue :: Issue -> IO ()
+assignToIssue issue = withOwnerRepo assignAction >>= \case
+    Left err              -> errorMessage (T.pack $ show err) >> exitFailure
```

Comment:
Should this be exitFailure? The new branch is already created at this point and the user is already checked out to it. I guess the meaningful message is better in here. What do you think?

Reply:
I like the idea of `exitFailure` because the command to the program was create a new branch **and** assign me to this issue. if we get a `Left` here the second part of that operation has failed, so I do think an indicate of failure is appropriate. Though I could see us just going with a warning instead. I'll leave it as is unless you feel strongly otherwise.

Reply:
I agree with @vrom911, and I think that the warning here is better. Basically, I don't see any way how `Left` can appear here. We need to handle this case because of the library API. But at this point, we already fetched the issue. It there's no such issue then the function will fail earlier. Otherwise, you should be assigned to the issue in every case. But if your internet connection was lost during this command execution, then a warning is more appropriate. I think of `--assign` as not as a separate command, but an addition to existing command. So `hit new` should fail only if `new` fails, not `--assign`. `--assign` is more like an additional plugin to `new`.